Make await expression in default args of async function a syntax error

ianwjhalliday opened the pull request:
Also improve the error message for when an await expression is used in
an invalid context.

msftclas commented:
Hi __@ianwjhalliday__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Ian Halliday). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

ianwjhalliday commented:
@boingoing @abchatra this one is a small quick review

akroshg commented on lib/Parser/parse.cpp:
```diff
@@ -7236,7 +7237,7 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
                 //
                 // Otherwise it is an error for a yield to appear in the context of a higher level
```
nit: 'await' - instead of yield.

ianwjhalliday replied:
Oh I meant to remove that comment since it was incorrect (from copy paste). Thanks, I'll remove it.

aneeshdk commented:
LGTM.

akroshg commented:
LGTM as well.

ianwjhalliday commented:
@aneeshdk @akroshg I fixed up the comments and added a couple new tests in a new commit.  Mind taking a look?  I will rebase into one commit after review.

akroshg commented:
LGTM.

dilijev commented on test/es7/asyncawait-syntax.js:
```diff
@@ -128,6 +128,25 @@ var tests = [
         }
     },
+    {
+        name: "await is a future reversed keyword and recognized in strict mode as an error in non-async functions",
```
future **reserved** keyword

ianwjhalliday replied:
ha whoops, thanks for catching that

dilijev commented:
Otherwise LGTM

ianwjhalliday commented:
Rebased and resolved merge conflict
